src/test/java/SpringJPA/UserPageTest.java

```diff
@@ -107,4 +110,14 @@ public void testPremiumUser() throws Exception{
     public void testAdminUser() throws Exception{
         this.mockMvc.perform(get("/user")).andDo(print()).andExpect(status().isOk());
     }
+
+    @Test
+    @WithMockUser(username="User1")
+    @DirtiesContext
+    public void testAfterTrial() throws Exception{
+        User loadedUser = userRepository.findByUsername("User1");
+        loadedUser.setStartTime(System.currentTimeMillis() - 2592000001l);
+        this.userRepository.save(loadedUser);
+        this.mockMvc.perform(get("/user")).andDo(print()).andExpect(status().isForbidden());
```

Comment:
You'll need to add a start trial time here

Reply:
That works as it is. setStartTime() and startTrial() both set the starttime parameter in user. So wrong place, but I did resolve the error in another location